mslib/msui/multiple_flightpath_dockwidget.py

```diff
@@ -92,177 +92,132 @@ def __init__(self, parent=None, view=None, listView=None, activeFlightTrack=None
 
         # Connect Signals and Slots
-        self.bt_addFile.clicked.connect(self.get_file)
-        self.btRemove_file.clicked.connect(self.remove_item)
-        self.listView.itemChanged.connect(self.syncListViews)
-        self.listView.itemChanged.connect(self.drawInactiveFlighttracks)
-        self.listView.itemActivated.connect(self.get_active_flighttrack)
+        self.list_flighttrack.itemChanged.connect(self.wait2)
+        self.listView.model().rowsInserted.connect(self.wait)
+        self.listView.model().rowsRemoved.connect(self.flighttrackRemoved)
+        self.listView.itemActivated.connect(self.activate_flighttrack)
+        # Set flags
+        self.flag = 0
+        self.flag1 = 0
+
+        # Load default flighttrack.
+        for index in range(self.listView.count()):
+            wp_model = self.listView.item(index).flighttrack_model
+            listItem = self.create_list_item(wp_model)
 
-        self.view.plot_multiple_flightpath(self)
+            self.save_waypoint_model_data(wp_model)
 
-        # Update Flighttrack list from MSUI list
-        self.syncListViews()
+            self.activate_flighttrack(listItem)
 
-    def syncListViews(self):
-        self.list_flighttrack.clear()
-        for index in range(self.listView.count()):
-            wp_model = self.listView.item(index).flighttrack_model
-            msui.QFlightTrackListWidgetItem(wp_model, self.list_flighttrack)
+    def update(self):
+        for entry in self.dict_files.values():
+            entry["patch"].update()
+
+    def remove(self):
+        for entry in self.dict_files.values():
+            entry["patch"].remove()
+
+    def wait(self, parent, start, end):
+        t1 = threading.Timer(1.0, self.flighttrackAdded, [parent, start, end])
+        t1.start()
+
+    def wait2(self):
+        t2 = threading.Timer(2.0, self.flagop)
+        t2.start()
+
+    def flagop(self):
+        if self.flag == 1:       # New flighttrack added
+            self.flag = 0
+        elif self.flag1 == 1:    # Flighttrack activated
+            self.flag1 = 0
+        else:
+            self.drawInactiveFlighttracks()
 
-    def get_file(self):
+    def flighttrackAdded(self, parent, start, end):
         """
-        Slot that open a file dialog to choose FTML file.
+        Slot to add flighttrack.
         """
-        filenames = get_open_filenames(
-            self, "Open FTML File", os.path.dirname(str(self.directory_location)), "FTML Files(*.ftml)")
-        if not filenames:
-            return
-        self.select_file(filenames)
+        self.flag = 1
+        wp_model = self.listView.item(start).flighttrack_model
+        listItem = self.create_list_item(wp_model)
+        self.save_waypoint_model_data(wp_model)
 
-    def select_file(self, filenames):
+        self.drawInactiveFlighttracks()
+        self.activate_flighttrack(listItem)
+
+    def save_waypoint_model_data(self, wp_model):
+        lst = []
```

Comment:
better name please,  or may be a list comprehension?

Reply:
I'll change it